contracts/strategies/allocation/QFOffchainCalc.sol

```diff
@@ -0,0 +1,194 @@
+    uint256 public applicationEndTime;
+    uint256 public votingStartTime;
+    uint256 public votingEndTime;
+
+    bool public identityRequired;
+
+    uint256 private _index;
+
+    Allo allo;
+
+    error APPLICATIONS_NOT_OPEN();
+    error IDENTITY_REQUIRED();
+    error VOTING_NOT_OPEN();
+
+    event ApplicationCreated(address indexed applicant, uint256 indexed applicationId);
+    event ApplicationStatusUpdated(address indexed applicant, uint256 indexed applicationId, Status status);
+    event ApplicationVoted(
+        address indexed voter, uint256 indexed applicationId, uint256 vote, address token, address recipient
+    );
+
+    /// @notice Struct to hold details of an application
+    struct Application {
+        address recipient;
+        uint256 applicationId;
+        uint256 amount;
+        uint256 applicationTime;
+        uint256 votingTime;
+        uint256 votes;
+        Status status;
+    }
+
+    // Note: this is mapped to the Allo global status's in the mapping below.
+    /// @notice Enum for the local status of the application
+    enum Status {
+        PENDING,
+        ACCEPTED,
+        REJECTED,
+        REAPPLIED
+    }
+
+    /// @notice recipient -> Status -> ApplicationStatus
+    mapping(address => mapping(Status => ApplicationStatus)) public globalStatusMap;
+
+    /// @notice recipient -> Application
+    mapping(address => Application) public applications;
+
+    /// @notice applicationId -> Application
+    mapping(uint256 => Application) public applicationIdToApplication;
+
+    constructor(address _alloCoreContract) {
+        allo = Allo(_alloCoreContract);
+    }
+
+    /// @notice Returns the owner identity
+    function getOwnerIdentity() external view override returns (string memory) {}
+
+    /// @notice Applies to the pool
+    /// @param _data The data to be decoded
+    /// @param sender The sender of the transaction
+    function applyToPool(bytes memory _data, address sender) external payable override returns (uint256) {
+        // todo: check that the application is valid and dates are open to apply
+        if (applicationStartTime > block.timestamp || applicationEndTime < block.timestamp) {
+            revert APPLICATIONS_NOT_OPEN();
+        }
+
+        if (identityRequired) {
+            // Todo: check that the user has an identity and revert if not
+            // revert IDENTITY_REQUIRED();
+        }
+
+        Application memory application = Application({
+            applicationId: ++_index,
+            recipient: sender,
+            amount: msg.value, // pull from _data
+            applicationTime: 0,
+            votingTime: 0,
+            votes: 0,
+            status: Status.PENDING
+        });
+
+        // Note: not sure where to grab the poolId here?
+        // allo.applyToPool(_poolId, _data);
+
+        // Set the status's of the application
+        globalStatusMap[sender][Status.PENDING] = IAllocationStrategy.ApplicationStatus.Pending;
```

Comment:
Might be missing this somewhere, but I don't see a local status map. How are you thinking these will play together?